api/views/recommendation.py

```diff
@@ -0,0 +1,113 @@
+import random
+
+from django.db import models
+from django.db.models import Count, Q
+from rest_framework import mixins, viewsets
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from api.models import Song, SongEntry, WishlistSong
+from api.serializers.recommendation import RecommendationSerializer
+from api.views.authentication.token import TokenBasedAuthentication
+
+
+class RecommendationViewSet(
+    mixins.RetrieveModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    API endpoint that constructs and serves a list of 6 songs: Wishlist > Likes > random recommendations.
+    """
+
+    serializer_class = RecommendationSerializer
+    queryset = WishlistSong.objects.all()
+    authentication_classes = [
+        SessionAuthentication,
+        TokenBasedAuthentication,
+    ]
+
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
+
+    def retrieve(self, request, *args, **kwargs):
+        session_key = self.kwargs["pk"]
+
+        wishes_list = list(
+            WishlistSong.objects.filter(user=request.user).order_by("?")[0:6]
+        )
+        if len(wishes_list) < 6:
+            self._get_six_recommendations(request, wishes_list, session_key)
+
+        results = [item for item in wishes_list][0:6]
+
+        serializer = self.get_serializer(results, many=True)
+        return Response(serializer.data)
+
+    def _get_six_recommendations(self, request, wishes_list, session_key):
+        likes_list = list(
+            Song.objects.filter(likes=request.user)
+            .exclude(song_entry__songbook__session_key=session_key)
+            .order_by("?")
+            .values("artist", "title")[0:6]
+        )
+
+        used_songs_entry = list(
+            SongEntry.objects.filter(songbook__session_key=session_key)
+        )
+
+        first_hundred = (
```

Comment:
Maybe change to `top_hunderd` instead? VS code should refactor it for you, right click variable name -> "Rename Symbol" -> enter new variable name

Reply:
I appreciate the suggestion! Done!